### ISSUE TITLE
r10k  deploy:  failed while running: private method `gsub' called for nil:NilClass

### ISSUE THREAD

**Issue by larstobi** [puppetlabs/r10k#25]
Using r10k version from git: b98b173a9c80dc92c9c2a1fb7f051ad9da50b1d4

```
root@master:~/r10k-shared# r10k  deploy display
[R10K::TaskRunner - ERROR] Task #<R10K::Task::Deployment::Display:0x7fc67a899890> failed while running: private method `gsub' called for nil:NilClass
root@master:~/r10k-shared# 
root@master:~/r10k-shared# r10k  deploy environment
[R10K::TaskRunner - ERROR] Task #<R10K::Task::Deployment::DeployEnvironments:0x7f8fb8f8c830> failed while running: private method `gsub' called for nil:NilClass
[R10K::TaskRunner - ERROR] Task #<R10K::Task::Deployment::PurgeEnvironments:0x7f8fb8f8c7b8> failed while running: private method `gsub' called for nil:NilClass
root@master:~/r10k-shared# r10k  deploy module
[R10K::TaskRunner - ERROR] Task #<R10K::Task::Deployment::DeployModules:0x7f4298d537e8> failed while running: private method `gsub' called for nil:NilClass
```

The contents of r10k.yaml:

```
root@master:~/r10k-shared# cat /etc/r10k.yaml
:cachedir: /var/cache/r10k
:sources:
  :master:
    remote:  'git@github.com:larstobi/r10k-shared.git'
    basedir: '/etc/puppetlabs/puppet/environments'
:purgedirs:
  - '/etc/puppetlabs/puppet/environments'
```

**Comment by larstobi** [comment]
I had an error in my r10k.yaml. Missing a colon in front of :remote.

```
:sources:
  :master:
    :remote:  'git@github.com:larstobi/r10k-shared.git'
    :basedir: '/etc/puppetlabs/puppet/environments'
```

Maybe an error message could help others if their :remote is missing or misspelled.

**Comment by adrienthebo** [comment]
The configuration format and validation has been a pain, and fixing it is on the roadmap for 1.0.0 per GH-18. However throwing a NilClass error in this case is pretty terrible, and this should have better validation. I'm reopening this to make sure that this reports a more meaningful error.

**Comment by adrienthebo** [comment]
Released in 1.0.0rc1
